src/Common/src/Interop/Unix/libcurl/Interop.libcurl.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libcurl
+    {
+        [DllImport(Interop.Libraries.LibCurl, SetLastError = true)]
+        public static extern SafeCurlHandle curl_easy_init();
+
+        [DllImport(Interop.Libraries.LibCurl, SetLastError = true)]
+        public static extern void curl_easy_cleanup(
+            IntPtr handle);
+
+        [DllImport(Interop.Libraries.LibCurl, CharSet = CharSet.Ansi, SetLastError = true)]
+        public static extern int curl_easy_setopt(
+            SafeCurlHandle curl,
+            int option,
+            string value);
+
+        [DllImport(Interop.Libraries.LibCurl, SetLastError = true)]
+        public static extern int curl_easy_setopt(
```

Comment:
Well, it'll cause additional work to be done, and the previous errno value cleared.  Plus if you're not going to use the error by querying Marshal.GetLastWin32Error(), it doesn't matter whether it sets the error or not.  The code you've written assumes that the result from the functions contains the error value (e.g. passing it to the function that converts the result to a string error description), so it would seem that at least from your current usage, these aren't needed.